pkg/controller/tiers/tiers_controller.go

```diff
@@ -136,7 +148,37 @@ func (r *ReconcileTiers) Reconcile(ctx context.Context, request reconcile.Reques
 		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
 	}
 
-	component := tiers.Tiers(&tiers.Config{Openshift: r.provider == operatorv1.ProviderOpenShift})
+	tiersConfig := tiers.Config{
+		Openshift:    r.provider == operatorv1.ProviderOpenShift,
+		NodeLocalDNS: false,
+		KubeDNSCIDR:  "0.0.0.0/32",
+	}
+
+	if r.provider != operatorv1.ProviderOpenShift {
+		nodeLocalDNSExists, _ := utils.IsNodeLocalDNSAvailable(ctx, r.client)
+		if err != nil {
+			r.status.SetDegraded(operatorv1.ResourceNotFound, "Node Local DNS not enabled", err, reqLogger)
+		} else if nodeLocalDNSExists {
+			// discover kube-dns ip address
+			k8sClient, _ := kubernetes.NewForConfig(r.config)
+			kubeDNSService, _ := k8sClient.CoreV1().Services("kube-system").Get(ctx, "kube-dns", metav1.GetOptions{})
```

Comment:
Is it possible to query Services using the client provided by the controller runtime (i.e. the Manager client)? 

Motivation being: this Get will not be cached, and will hit the API server every reconcile. The manager client maintains caches for resources